cfme/tests/intelligence/test_reports_with_timelines.py

```diff
@@ -1,16 +1,297 @@
+                          "Event Type",
+                          "Source",
+                          "Parent Host",
+                          "Date Created"
+                          ],
+        "filter": {"primary_filter": "fill_field(field=Management Event : Date Created, "
+                                     "value=This Week);"},
+        "timeline": {"based_on": "Date Created"}
+    }
+    report = appliance.collections.reports.create(**param_dict)
+    yield report
+    if report.exists:
+        report.delete()
+
+
+@pytest.fixture
+def report_power_operations(appliance):
+    param_dict = {
+        "menu_name": "VMs Powered On/Off this week",
+        "title": "VMs Powered On/Off this week",
+        "base_report_on": "Management Events",
+        "cancel_after": "2 Hours",
+        "report_fields": ["VM Name",
+                          "Event Type",
+                          "Source",
+                          "Parent Host",
+                          "Date Created"
+                          ],
+        "filter": {"primary_filter": "fill_field(field=Management Event : Activity Sample - "
+                                     "Timestamp (Day/Time), key=IS, value=This Week);"},
+        "timeline": {"based_on": "Date Created"},
+
+    }
+    report = appliance.collections.reports.create(**param_dict)
+    yield report
+    if report.exists:
+        report.delete()
+
+
+@pytest.fixture
+def report_hosts_data_management(appliance):
+    param_dict = {
+        "menu_name": "host data management",
+        "title": "host data management",
+        "base_report_on": "Management Events",
+        "cancel_after": "2 Hours",
+        "report_fields": ["VM Name",
+                          "Event Type",
+                          "Source",
+                          "Parent Host",
+                          "Date Created"
+                          ],
+        "filter": {"primary_filter": "fill_field(field=Management Event.Host / Node : Date Created,"
+                                     "key=IS, value=This Month);"},
+        "timeline": {"based_on": "Date Created"}
+    }
+    report = appliance.collections.reports.create(**param_dict)
+    yield report
+    if report.exists:
+        report.delete()
+
+
+@pytest.mark.meta(blockers=[BZ(1564454, forced_streams=['5.8'])])
+def test_custom_report_policy_timeline(new_vm, control_policy, report_policy_event_week):
+    """Create a policy event and assign it to the VM, stop the vm and verify that the policy
+    policy events appears in the cloud intel report timelines. the report is a copy of one
+    existing in Reports"""
+    logger.info('Will stop VM %r', new_vm.name)
+    wait_for(new_vm.provider.mgmt.stop_vm, [new_vm.name], timeout='7m',
+             message='Stopping VM failed')
+    view = navigate_to(report_policy_event_week, 'Timelines')
+    assert view.is_timeline_displayed, 'Timeline is not displayed.'
+    view.flash.assert_no_error()
+
+    def check_timeline():
```

Comment:
REQUIRED: you have this procedure in every test. it is almost the same. I think you need to define it once and use in all functions.

Reply:
fixed